setupmeta/hook.py

```diff
@@ -7,22 +7,39 @@
 from setupmeta.model import MetaDefs, SetupMeta
 
 
-# Reference to original distutils.dist.Distribution.parse_command_line
-dd_original = distutils.dist.Distribution.parse_command_line
+finalize_options_orig = distutils.dist.Distribution.finalize_options
+def finalize_options(dist):  # noqa: E302 (keep override close to function it replaces)
+    """
+    Hook into setuptools' Distribution class before attributes are interpreted.
+
+    This is called before Distribution attributes are finalized and validated,
+    allowing us to transform attribute values before they have to conform to
+    the usual spec. This step is *before* configuration is additionally read
+    from config files.
+    """
+    dist._setupmeta = SetupMeta().preprocess(dist)
```

Comment:
`SetupMeta()` used to be like a singleton before.
Should be OK, but possibly simplifiable